quil-rs/src/validation/identifier.rs

```diff
@@ -0,0 +1,34 @@
+//! Types and functions related to validating Quil identifiers
```

Comment:
In order to have a "source of truth" for validation rules across the parser and constructors for different instruction types, I opted to create a validation module.